### ISSUE TITLE
`open_for_directories = true` not working as expected with `Neo-tree` enabled and `hijack_netrw_behavior = "disabled"`

### ISSUE THREAD

**Issue by xfzv** [mikavilpas/yazi.nvim#308]
```console
% yazi -V
Yazi 0.3.0 (7a380c2 2024-08-01)
```

With the following `repro.lua`:

```lua
-- You can use this file to reproduce an issue with your configuration.

---@module "yazi"
---@module "lazy"

-- DO NOT change the paths and don't remove the colorscheme
local root = vim.fn.fnamemodify("./.repro", ":p")

-- set stdpaths to use .repro
for _, name in ipairs({ "config", "data", "state", "cache" }) do
  vim.env[("XDG_%s_HOME"):format(name:upper())] = root .. "/" .. name
end

-- bootstrap lazy
local lazypath = root .. "/plugins/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
  vim.fn.system({
    "git",
    "clone",
    "--filter=blob:none",
    "https://github.com/folke/lazy.nvim.git",
    lazypath,
  })
end
vim.opt.runtimepath:prepend(lazypath)
vim.g.mapleader = " "

-- install the following plugins
---@type LazySpec
local plugins = {
  "folke/tokyonight.nvim",
  {
    "mikavilpas/yazi.nvim",
    event = "VeryLazy",
    keys = {
      {
        -- 👇 choose your own keymapping
        "<leader>fy",
        function()
          require("yazi").yazi()
        end,

        { desc = "Open the file manager" },
      },
    },
    ---@type YaziConfig
    config = function()
      require("yazi").setup({
        open_for_directories = true,
      })
    end,
  },
  {
    "nvim-neo-tree/neo-tree.nvim",
    branch = "v3.x",
    cmd = "Neotree",
    dependencies = {
      "nvim-lua/plenary.nvim",
      "nvim-tree/nvim-web-devicons",
      "MunifTanjim/nui.nvim",
    },
    opts = {
      filesystem = {
        hijack_netrw_behavior = "disabled",
      },
    },
  },
}
require("lazy").setup(plugins, {
  root = root .. "/plugins",
})

vim.cmd.colorscheme("tokyonight")
-- add anything else here
```

This is what happens when trying to open the current directory with:

```console
% nvim -u repro.lua .
```

https://github.com/user-attachments/assets/4e2d164c-99b4-4025-9848-8918ef7ed517

- `newtr` is displayed for a few seconds
- the following error is displayed: `/tmp/repro/repro//plugins/lazy.nvim/lua/lazy/core/handler/event.lua:161: Invalid buffer id: 2`
- `nextr` disappears and `yazi` appears

With my own config, I'm still getting the error (getting `id: 1` instead of `id: 2`) but `yazi` shows up immediately, I'm not seeing `newtr` at all.

**Comment by mikavilpas** [comment]
Thanks for the reproduction. I am able to reproduce this and will try to find a fix.

**Comment by mikavilpas** [comment]
By the way, I would recommend changing your yazi.nvim lazy spec to the following:

```lua
  {
    'mikavilpas/yazi.nvim',
    dir = vim.fn.expand('~/git/yazi.nvim/'),
    event = 'VeryLazy',
    keys = {
      {
        -- 👇 choose your own keymapping
        '<leader>fy',
        function()
          require('yazi').yazi()
        end,

        { desc = 'Open the file manager' },
      },
    },
    ---@type YaziConfig
    config = {
      open_for_directories = true,
    },
  },
```

In your reproduction, it seems to have a function where a table is expected. I don't think it has any bad effects right now though, but in the future there might be something (that I can't think of right now) 🙂

**Comment by mikavilpas** [comment]
Oops, perhaps it's best to ignore the `dir` key in what I just pasted - I used it myself when investigating this.

**Comment by mikavilpas** [comment]
> Can confirm it's fixed on my end with #309 (using `opts = {}` instead of calling `yazi.setup()`)

Awesome! I'm not sure what broke it but the important thing is that it's now fixed again 😄

**Comment by mikavilpas** [comment]
> I'm actually using `opts = {}` in my personal config but while searching for a possible solution before opening this issue, I found
> 
> https://github.com/mikavilpas/yazi.nvim/blob/3fbb0a989473149144ede43d2e7be0deff687b2d/README.md?plain=1#L157-L159
> 
> So that's why I tried with
> 
> ```lua
> config = function()
>       require("yazi").setup({
>         open_for_directories = true,
>       })
>     end,
> ```
> 
> in the `repro.lua` file.
> 
> Maybe I misunderstood and sticking with `opts = {}` is fine?

I think lazy.nvim calls `.setup()` if `opts` are given. It's a bit confusing but mostly works in my opinion, hehe

![image](https://github.com/user-attachments/assets/88a348ae-5a90-4314-b368-7ce0ec719e57)
